### DISCUSSION TITLE
Handle logout and user-dependent queries

### DISCUSSION THREAD

**Discussion by bacarybruno** [TanStack/query#7839] [Q&A] [answered]
Heyo all 👋

We are currently developing a React Native app that uses TanStack Query. This app is available in **multiple countries**, and users can have **accounts in different countries**. Additionally, users can use the app **without being logged in**.

We have several queries in the app that depend on the connected user. But we are not directly sending the user ID in the query parameters: the backend identifies the user based on cookies.

I need your recommendations on how we should handle query keys and user login / logout / change country etc. We have a few potential solutions:
1️⃣ Add `userId` and `country` to each query key: This ensures data is refetched whenever the user or country changes
2️⃣ Prefix query keys with something like `['ME', ...]`: this allows us to invalidate all queries that depend on a specific user
3️⃣ Clear the React Query cache or invalidate all queries on logout
4️⃣ Put a lower `gc_time`: can be done in combinaison with other solutions maybe?
5️⃣  Insert your solution here

What do you think would be the best practice for our case?
Thanks for your help 🙏

**Comment by TkDodo** [comment, marked as answer]
1 and 3 together sounds good
